In [1]:
from pandas import read_excel

**Part A: Filtering the data using '.query()'**

In [2]:
city_data = read_excel("MA-Jan-14-2021-city-data.xlsx")
city_data

City/Town Total Case Counts Two Week Case Counts Average Daily Rate  \
0       Abington              1073                  224          87.795166   
1          Acton               596                   90          27.093016   
2       Acushnet               808                  241         164.741333   
3          Adams               171                   46           39.93607   
4         Agawam              1699                  288          72.190166   
..           ...               ...                  ...                ...   
348  Worthington                20                   <5           6.823981   
349     Wrentham               671                  106          67.174173   
350     Yarmouth               772                  140          41.559829   
351      Unknown              1290                  256                  *   
352        State            427752                76054                 78   

    % Change in Last Week  Total Tests  Total Tests Last Two Weeks  \
0                  Higher        18095                        2064   
1                  Higher        25882                        2510   
2                  Higher        12928                        1893   
3                  Higher         8095                         991   
4                  Higher        38404                        3965   
..                    ...          ...                         ...   
348             No Change         1158                          98   
349                Higher        18750                        1936   
350                 Lower        23531                        2862   
351                     *       209782                       16941   
352                Higher     11944985                     1069689   

     Total Positive Tests Percent Positivity Change Since Last Week  
0                     240           0.116279                  Lower  
1                     100           0.039841              No Change  
2                     272           0.143687                 Higher  
3                      48           0.048436                 Higher  
4                     319           0.080454                 Higher  
..                    ...                ...                    ...  
348                     1           0.010204                  Lower  
349                   114           0.058884                 Higher  
350                   162           0.056604                  Lower  
351                   259                  *                      *  
352                 85470             0.0799                 Higher  

[353 rows x 10 columns]

In [3]:
# city_data.columns = [column.replace(" ", "_") for column in city_data.columns]

In [4]:
filtered = city_data.query('`Total Case Counts` != "<5" and `Percent Positivity` != "*"')
filtered = filtered[['Total Case Counts', 'Percent Positivity']]

In [5]:
filtered

Total Case Counts Percent Positivity
0                1073           0.116279
1                 596           0.039841
2                 808           0.143687
3                 171           0.048436
4                1699           0.080454
..                ...                ...
347             16924           0.084394
348                20           0.010204
349               671           0.058884
350               772           0.056604
352            427752             0.0799

[346 rows x 2 columns]

There are 346 data instances in the new DataFrame.

**Part B: Determine how many cities or towns simultaneously had both more than 1000 cases and a positivity percentage less than .1**

In [6]:
filtered_partb = filtered.query('`Total Case Counts` > 1000 and `Percent Positivity` < .1')

In [7]:
filtered_partb

Total Case Counts Percent Positivity
4                1699           0.080454
8                1748           0.071254
10               1194           0.034736
14               1025           0.086225
16               2732           0.078713
17               1082            0.09375
20               2128           0.088017
30               2342           0.074249
31               2574           0.099944
35              46455           0.059139
40               2664           0.083113
46               1438           0.027907
48               1261           0.074861
49               3346           0.018063
50               1117           0.074295
56               1866           0.082755
71               2171           0.075089
73               1407           0.070344
85               1035           0.075984
100              5640           0.075118
101              1038           0.071893
106              1182           0.082225
130              1133           0.053716
136              3514           0.081138
166              1009           0.082383
175              3632           0.067546
177              1330           0.045224
188              1522           0.056898
197              1256           0.046924
198              1045            0.03953
206              2692             0.0296
209              2006           0.070026
210              1349           0.089709
219              1846           0.072667
235              1777            0.05813
238              2762           0.092811
242              4757           0.074628
245              1316           0.081476
257              3225            0.08236
270              1969           0.091367
273              3727           0.042958
283              1629           0.075609
294              2299             0.0788
304              1513           0.085372
306              1171           0.083556
307              3716           0.072045
313              1563           0.049099
327              1007           0.040918
328              1949           0.091559
345              1626           0.062018
346              2941           0.088546
347             16924           0.084394
352            427752             0.0799

**Part C: Use pandas to print the mean and standard deviation of the positivity percentage**

In [8]:
mean = filtered['Percent Positivity'].mean()
mean

0.0766657688113855

In [9]:
stdev = filtered['Percent Positivity'].std()
stdev

0.038291167032190127

In [10]:
print("Mean and Standard Deviation (Percent Positivity_: {} +- {}".format(mean, stdev))

Mean and Standard Deviation (Percent Positivity_: 0.0766657688113855 +- 0.038291167032190127
